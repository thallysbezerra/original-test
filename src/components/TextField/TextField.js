import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './TextField.scss';

export default class TextField extends Component {
	constructor(props) {
		super(props);
		this.state = {
			value: ''
		};
	}

	handleChange = event => {
		this.setState({ value: event.target.value });
	};

	render() {
		const {
			className,
			icon,
			id,
			onClickIcon,
			placeholder,
			type
		} = this.props;

		const { value } = this.state;

		return (
			<div
				className={`text-field ${className ? className : ''} ${
					value !== '' ? 'active' : ''
				}`}
			>
				<input
					className={`text-field__input text-field__input--${
						!icon ? 'no-icon' : ''
					}`}
					id={id}
					onChange={this.handleChange}
					type={type}
					value={value}
				/>
				{icon && (
					<button className="text-field__button" onClick={onClickIcon}>
						<i className={`text-field__button__icon fas fa-${icon}`} />
					</button>
				)}
				<label className="text-field__label">{placeholder}</label>
			</div>
		);
	}
}

TextField.propTypes = {
	className: PropTypes.string,
	icon: PropTypes.string,
	id: PropTypes.string,
	onClickIcon: PropTypes.func,
	placeholder: PropTypes.string.isRequired,
	type: PropTypes.string
};

TextField.defaultProps = {
	type: 'text'
};
