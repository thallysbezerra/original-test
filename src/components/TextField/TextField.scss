@import '../../css/Theme.scss';

.text-field {
	position: relative;
	width: 100%;

	&.active {
		.text-field {
			&__label {
				font-size: 12px;
				transform: translateY(-18px);
			}
		}
	}

	&__input {
		border: none;
		border-bottom: 1px solid $grey-four;
		box-sizing: border-box;
		color: $grey-seven;
		height: 32px;
		margin-bottom: 20px;
		outline: none;
		padding: 0 0 0 50px;
		transition: border-color 0.15s ease-in-out;
		width: 100%;

		&:focus {
			border-color: $salmon;

			& ~ .text-field__label {
				color: $salmon;
				font-size: 12px;
				transform: translateY(-18px);
			}
		}

		& + .text-field__label {
			left: 0;
		}

		&--no-icon {
			padding: 0;
		}
	}

	&__label {
		color: $grey-seven;
		left: 50px;
		position: absolute;
		top: 10px;
		transition: color 0.15s ease-in-out, font-size 0.15s ease-in-out,
			transform 0.15s ease-in-out;
	}

	&__button {
		align-items: center;
		background-color: transparent;
		border: none;
		color: $grey-seven;
		display: flex;
		font-size: 21px;
		height: 32px;
		justify-content: center;
		outline: none;
		position: absolute;
		transition: color 0.15s ease-in-out;
		top: 0;
		width: 32px;

		&:hover {
			color: $salmon;
			cursor: pointer;
		}
	}
}
