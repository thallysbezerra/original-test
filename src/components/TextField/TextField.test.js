import React from 'react';
import TextField from './TextField';

import renderer from 'react-test-renderer';

describe('TextField UI', () => {
	it('<TextField /> must have the snapshot version.', () => {
		const tree = renderer.create(<TextField />);

		expect(tree.toJSON()).toMatchSnapshot();
	});
});
