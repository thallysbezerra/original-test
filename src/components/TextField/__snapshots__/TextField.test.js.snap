// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField UI <TextField /> must have the snapshot version. 1`] = `
<div
  className="text-field  "
>
  <input
    className="text-field__input text-field__input--no-icon"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    className="text-field__label"
  >
    Needs a placeholder
  </label>
</div>
`;
